### PR TITLE
CAD-2772 making tx-generator node-ready

### PR DESCRIPTION
**Pull request by MarcFontaine**
Updates for the `tx-generator` to make it read for `cardano-node`.
Fixes suggested by hlint.

### CODE REVIEW

File: cardano-tx-generator/cardano-tx-generator.cabal [hunk 1]
```diff
@@ -21,8 +21,8 @@ library
                         -Wno-unticked-promoted-constructors
                         -Wpartial-fields
                         -Wredundant-constraints
-                        -threaded -rtsopts -with-rtsopts=-N -with-rtsopts=-T
-
+                        -Wwarn=deprecations
+
```
Comment (deepfire) [inline cardano-tx-generator/cardano-tx-generator.cabal]: This seems like a whitespace-only change.

File: cardano-tx-generator/src/Cardano/Benchmarking/GeneratorTx/Tx.hs [hunk 2]
```diff
@@ -182,7 +180,7 @@ mkTransactionGen signingKey inputs address payments metadata fee =
   offsetMap = Map.fromList $ zipWith (\payment index -> (fst payment, TxIx index))
                                      payments
                                      [0..]
-  txOutSum :: [(TxOut era)] -> Lovelace
+  txOutSum :: [ TxOut era ] -> Lovelace
```
Comment (deepfire) [inline cardano-tx-generator/src/Cardano/Benchmarking/GeneratorTx/Tx.hs:183]: The extra space is fine with me, but can be perceived as odd, FWIW..

File: cardano-tx-generator/test/script.json [hunk 3]
```diff
@@ -84,6 +84,13 @@
         ]
     },
     {
-        "runBenchmark": "txlist"
-    }
+        "asyncBenchmark": "thread1",
+        "txList": "txlist"
+    },
+    {
+        "waitForEra": "Byron"
```
Comment (deepfire) [inline cardano-tx-generator/test/script.json]: Wait for.. Byron?  I think I'm missing something..

What does that mean?

Comment (MarcFontaine) [reply cardano-tx-generator/test/script.json]: Right that does not really make sense.
Though the effect would be wait forever...
This is just the 'testscript' for testing the ToJSON and FromJSON instances.
but this just reminds me.. the real script needs to be updated

File: cardano-tx-generator/cardano-tx-generator.cabal [hunk 4]
```diff
@@ -21,7 +21,7 @@ library
                         -Wno-unticked-promoted-constructors
                         -Wpartial-fields
                         -Wredundant-constraints
-                        -threaded -rtsopts -with-rtsopts=-N -with-rtsopts=-T
+                        -Wwarn=deprecations
```
Comment (deepfire) [inline cardano-tx-generator/cardano-tx-generator.cabal:24]: @MarcFontaine, what is the expected impact of this change?

**Review by deepfire — APPROVED** [review]
LGTM!

**Pull request merged**